### PR TITLE
[CGALPlugin] new features

### PR DESCRIPTION
**Pull request by EulalieCoevoet**
Adds new features with examples:
- BooleanOperations
image
- UpsamplePointCloud
image
- PoissonSurfaceReconstruction
image 
- FrontSurfaceReconstruction
image

Fixes compilation in CylinderMesh:
- **EDIT**: removed unused `#include <sofa/gl/template.h>`

Minor cleaning in DecimateMesh:
- Data member variables names with d_

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

File: applications/plugins/CGALPlugin/CMakeLists.txt [hunk 1]
```diff
@@ -51,7 +59,7 @@ if(CGAL_VERSION VERSION_GREATER 4.9) # if CGAL >= 4.10
 endif()
 
 # Link the plugin library to its dependencies (other libraries).
-target_link_libraries(${PROJECT_NAME} PUBLIC SofaCore SofaSimulationCommon)
+target_link_libraries(${PROJECT_NAME} PUBLIC SofaCore SofaSimulationCommon Sofa.GL)
```
Comment (fredroy) [inline applications/plugins/CGALPlugin/CMakeLists.txt]: > 
> Fixes compilation:
> 
>     * Sofa.GL
> 

This issue was fixed in #2065 so you dont need to add Sofa.GL

**Review by fredroy — CHANGES_REQUESTED** [review]

**Review by fredroy — APPROVED** [review]

File: applications/plugins/CGALPlugin/scenes/UpsamplePointCloud.scn [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+		<MechanicalObject showObject="1" showObjectScale="5" showColor="1 0 0"/>
+	</Node>
+
+	<Node name="UpsamplePointCloud">
+		<UpsamplePointCloud name="engine" src="@../PointCloud/loader"/>
+		<MeshTopology name="topology" position="@engine.outputPosition"/>
+		<MechanicalObject src="@topology" showObject="1" showObjectScale="5"/>
+	</Node>
+
+	<!-- <Node name="FrontSurfaceReconstruction">
```
Comment (epernod) [inline applications/plugins/CGALPlugin/scenes/UpsamplePointCloud.scn]: could you remove commented code

File: applications/plugins/CGALPlugin/src/CGALPlugin/DecimateMesh.h [hunk 3]
```diff
@@ -98,18 +98,18 @@ class DecimateMesh : public sofa::core::DataEngine
     bool testVertexAndFindIndex(const Vec3 &vertex, int &index);
 
     //Inputs
-    sofa::core::objectmodel::Data<VecCoord> m_inVertices; ///< List of vertices
-    sofa::core::objectmodel::Data<SeqTriangles> m_inTriangles; ///< List of triangles
-    sofa::core::objectmodel::Data<int> m_edgesTarget; ///< Desired number of edges after simplification
-    sofa::core::objectmodel::Data<float> m_edgesRatio; ///< Ratio between the number of edges and number of initial edges
+    sofa::core::objectmodel::Data<VecCoord> d_inVertices; ///< List of vertices
```
Comment (epernod) [inline applications/plugins/CGALPlugin/src/CGALPlugin/DecimateMesh.h:101]: renaming the Data is good. We just need to put the breaking flag to this PR

**Review by epernod — CHANGES_REQUESTED** [review]
Thanks for the contribution with scenes example. 
It would be nice to put some pictures of the scene results either here in the PR description or in the plugin Doc. So we will be able to check in the future if the behavior is still the one expected from the component (unless you prefer to write some tests ;) ).

**Review by epernod — APPROVED** [review]

**Pull request merged**